File: Medium/integer_to_roman.go
package main

import (
	"fmt"
	"math"
)

func intToRoman(num int) string {
	// symbolMap := map[string]int{"M": 1000, "CM": 900, "D": 500, "CD": 400, "C": 100, "XC": 90, "L": 50, "XL": 40, "X": 10, "IX": 9, "V": 5, "IV": 4, "I": 1}
	symbolMap := []string{"M", "CM", "D", "CD", "C", "XC", "L", "XL", "X", "IX", "V", "IV", "I"}
	symbolVal := []int{1000, 900, 500, 400, 100, 90, 50, 40, 10, 9, 5, 4, 1}
	romanText := ""

	for symbolIndex, val := range symbolVal {
		if num >= val {
			totalSymbol := math.RoundToEven(float64(num / val))
			for i := 0; i < int(totalSymbol); i++ {
				romanText += symbolMap[symbolIndex]
				num -= val
			}
		}
	}

	return romanText
}

func run_intToRoman() {
	fmt.Println(intToRoman(3749)) //MMMDCCXLIX
	fmt.Println(intToRoman(58))   //LVIII
}

File: Medium/integer_to_roman/integer_to_roman.java
package integer_to_roman;

import java.util.LinkedHashMap;
import java.util.Map;

class Solution {
    public String intToRoman(int num) {
        Map<String, Integer> symbolMap = new LinkedHashMap<>();
        symbolMap.put("M", 1000);
        symbolMap.put("CM", 900);
        symbolMap.put("D", 500);
        symbolMap.put("CD", 400);
        symbolMap.put("C", 100);
        symbolMap.put("XC", 90);
        symbolMap.put("L", 50);
        symbolMap.put("XL", 40);
        symbolMap.put("X", 10);
        symbolMap.put("IX", 9);
        symbolMap.put("V", 5);
        symbolMap.put("IV", 4);
        symbolMap.put("I", 1);

        String romanText = "";

        for(Map.Entry<String, Integer> symbol : symbolMap.entrySet()) {
            if(num >= symbol.getValue()) {
                int totalSymbol = (int) Math.floor(num/symbol.getValue());
                for (int i = 0; i < totalSymbol; i++) {
                    romanText += symbol.getKey();
                    num -= symbol.getValue();
                }
            }
        }

        return romanText;
    }
}

public class integer_to_roman {
    public static void main(String[] args) {
        Solution solution = new Solution();

        System.out.println(solution.intToRoman(3749)); // MMMDCCXLIX
        System.out.println(solution.intToRoman(58)); // LVIII
    }
}
